### PR TITLE
MOTECH-2311 Motech can start without config-locations.properties file

### PR DESCRIPTION
**Pull request by martokarski**
config-locations.properties file is used to find motech properties files (bootstrap, datanucleus, ...)
If this file does not exist in default directory, bootstrap.properties cannot be saved.  
This error doesn't occure in maven installation, beacuse maven copies this file to destination directory.

I added check if file exist and if doesn't it is created.

### CODE REVIEW

**Comment by pgesek** [comment]
I'll try looking at this today/tomorrow, I swear

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java [hunk 1]
```diff
@@ -115,6 +117,31 @@ public static Properties getPropertiesFromSystemVarString(String string) {
     }
 
     /**
+     * Returns {@code PropertiesConfiguration} and creates config file if does not exist
+     * @param basePath path to the file with config
+     * @param fileName name of the file with config
+     */
+    public static PropertiesConfiguration getPropertiesConfiguration(String basePath, String fileName) {
```
Comment (pgesek) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: missleading name, this creates a configuration

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java [hunk 2]
```diff
@@ -115,6 +117,31 @@ public static Properties getPropertiesFromSystemVarString(String string) {
...
+        File configFile = new File(basePath, fileName);
+        try {
```
Comment (pgesek) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: should we check if this file exists?

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java [hunk 3]
```diff
@@ -115,6 +117,31 @@ public static Properties getPropertiesFromSystemVarString(String string) {
...
+            throw new MotechConfigurationException(String.format("Cannot create file %s", configFile.getAbsolutePath()), e);
+        }
+
+        PropertiesConfiguration propertiesConfiguration = new PropertiesConfiguration();
+        propertiesConfiguration.setBasePath(basePath);
+        propertiesConfiguration.setFileName(fileName);
+        try {
+            propertiesConfiguration.load();
+        } catch (ConfigurationException e) {
+            LOGGER.error("Unable to load config locations: " + e.getMessage());
```
Comment (pgesek) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: this should throw an exception, the same as above

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java (hunk 1 above)
Comment (pgesek) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: needs tests

**Comment by pgesek** [comment]
Sorry about the delay, this looks fine, except the comments. This is low priority for now

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java (hunk 1 above)
Comment (martokarski) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: added tests

Comment (martokarski) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: Main purpose of this method is to return PropertiesConfiguration. Create a file is only an utility method for this. I moved this creation to another method for readability.

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java (hunk 2 above)
Comment (martokarski) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: These methods create only if file/directory isn't currently created.

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java [hunk 4]
```diff
@@ -115,6 +117,39 @@ public static Properties getPropertiesFromSystemVarString(String string) {
...
+        }
+
+        return propertiesConfiguration;
+    }
+
+    private static void createFileIfDoesNotExist(String basePath, String fileName) {
+        File configFile = new File(basePath, fileName);
+
+        try {
+            new File(configFile.getParent()).mkdirs();
```
Comment (kmadej) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: please add this brief comment that this methods creates file only if a file with this name does not yet exist

File: platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java [hunk 5]
```diff
@@ -115,6 +117,39 @@ public static Properties getPropertiesFromSystemVarString(String string) {
     }
 
     /**
+     * Returns {@code PropertiesConfiguration} and creates file for this configuration if it does not exist
+     * @param basePath path to the file with config
+     * @param fileName name of the file with config
+     */
+    public static PropertiesConfiguration getPropertiesConfiguration(String basePath, String fileName) {
```
Comment (kmadej) [inline platform/config-core/src/main/java/org/motechproject/config/core/utils/ConfigPropertiesUtils.java]: rename method to createPropertiesConfiguration, adjust javadoc

**Comment by martokarski** [comment]
test this please

**Pull request merged**